### PR TITLE
[2.0] React to X509Chain changes in macOS 10.13.4

### PR DESCRIPTION
**Pull request by bartonjs**
10.13.4 changed some of the detail codes for building the X509ChainStatusFlag values.

"ValidLeaf" (etc) => "TemporalValidity"
"WeakLeaf" (etc) => "WeakKeySize"
new "MissingIntermediate" when the chain didn't complete instead of "AnchorTrusted" lower down.

Port #28673 to release/2.0.0

### CODE REVIEW

**Comment by bartonjs** [comment]
Hm, right.  Maybe I'll do the other merge resolution of dropping the test since it was to codify having fixed a Linux behavior in 2.1.

File: src/System.Security.Cryptography.X509Certificates/tests/ChainTests.cs [hunk 1]
```diff
@@ -5,6 +5,7 @@
 using System.Collections.Generic;
 using System.IO;
 using System.Linq;
+using System.Runtime.InteropServices;
```
Comment (kasper3) [inline src/System.Security.Cryptography.X509Certificates/tests/ChainTests.cs]: Unused `using` :)

Comment (bartonjs) [reply src/System.Security.Cryptography.X509Certificates/tests/ChainTests.cs]: Whoops, thanks :)

**Comment by Petermarcu** [comment]
Next steps? We should get this into 2.1 Preview2 for sure and quickly into a 2.0 update.

**Comment by bartonjs** [comment]
@Petermarcu The fix is already in master, and I believe we're merging master to release/2.1 today for preview2.  If that's true, I don't think there's anything else we really can/should do for 2.1.

**Comment by bartonjs** [comment]
@corretge I assume you meant to comment on #28652?  The original reporter there said that the fixed fixed it for him.  Though he took the binary directly.  It's possible the SDK hasn't pulled in the master version of this change yet.

Please continue discussion on #28652.

**Pull request merged**